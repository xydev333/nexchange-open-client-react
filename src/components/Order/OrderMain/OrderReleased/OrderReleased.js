import React, { Component } from 'react';
import _ from 'lodash';
import getBlockchainUrl from 'Utils/getBlockchainUrl';
import OrderLinks from '../OrderLinks/OrderLinks';

class OrderReleased extends Component {
  constructor(props) {
    super(props);

    this.coin = props.order.pair.base;
    this.minConfirmations = this.coin.min_confirmations;
    this.tx = _.find(props.order.transactions, { type: 'W' });
    this.txId = this.tx.tx_id;
  }

  UNSAFE_componentWillReceiveProps(nextProps) {
    this.tx = _.find(nextProps.order.transactions, { type: 'W' });
    this.txId = this.tx.tx_id;
  }

  render() {
    if (this.txId === '' || this.txId === null) {
      return (
        <div className="col-xs-12 text-center order-status-section">
          <h2 style={{ margin: '0' }}>Processing withdrawal...</h2>
          <OrderLinks {...this.props} />
        </div>
      );
    }

    return (
      <div className="col-xs-12 text-center order-status-section">
        <h2 style={{ margin: '0' }}>
          Funds released, awaiting confirmations ({this.tx.confirmations}/{this.minConfirmations})
        </h2>
        <h5>
          Transaction ID:{' '}
          <a href={getBlockchainUrl(this.coin.code, this.txId)} target="_blank" className="text-green">
            {this.txId}
          </a>
        </h5>

        <OrderLinks coin={this.coin.code} txId={this.txId} {...this.props} />
      </div>
    );
  }
}

export default OrderReleased;
