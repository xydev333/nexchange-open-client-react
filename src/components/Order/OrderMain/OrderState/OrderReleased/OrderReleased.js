import React, { Component } from 'react';
import _ from 'lodash';
import getBlockchainUrl from 'Utils/getBlockchainUrl';
import OrderLinks from '../OrderLinks/OrderLinks';
import styles from '../OrderState.scss';

class OrderReleased extends Component {
  constructor(props) {
    super(props);

    this.coin = props.order.pair.base;
    this.minConfirmations = this.coin.min_confirmations;
    this.tx = _.find(props.order.transactions, { type: 'W' });
    this.txId = this.tx.tx_id;
  }

  UNSAFE_componentWillReceiveProps(nextProps) {
    this.tx = _.find(nextProps.order.transactions, { type: 'W' });
    this.txId = this.tx.tx_id;
  }

  render() {
    if (this.txId === '' || this.txId === null) {
      return (
        <div className={styles.container}>
          <h2 className={styles.title}>Processing withdrawal...</h2>
          <OrderLinks {...this.props} />
        </div>
      );
    }

    return (
      <div className={styles.container}>
        <h2 className={styles.title}>
          Funds released, awaiting confirmations ({this.tx.confirmations}/{this.minConfirmations})
        </h2>
        <h3 className={styles.subtitle}>
          Transaction ID:{' '}
          <a href={getBlockchainUrl(this.coin.code, this.txId)} target="_blank" className="text-green">
            {this.txId}
          </a>
        </h3>

        <OrderLinks coin={this.coin.code} txId={this.txId} {...this.props} />
      </div>
    );
  }
}

export default OrderReleased;
