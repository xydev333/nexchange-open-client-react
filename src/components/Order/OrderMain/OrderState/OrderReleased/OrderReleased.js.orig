import React, { Component } from 'react';
import { I18n } from 'react-i18next';
import _ from 'lodash';
import getBlockchainUrl from 'Utils/getBlockchainUrl';
import OrderLinks from '../OrderLinks/OrderLinks';
import styles from '../OrderState.scss';

class OrderReleased extends Component {
  constructor(props) {
    super(props);

    this.coin = props.order.pair.base;
    this.minConfirmations = this.coin.min_confirmations;
    this.tx = _.find(props.order.transactions, { type: 'W' });
    this.txId = this.tx.tx_id;
  }

  UNSAFE_componentWillReceiveProps(nextProps) {
    this.tx = _.find(nextProps.order.transactions, { type: 'W' });
    this.txId = this.tx.tx_id;
  }

  render() {
    if (this.txId === '' || this.txId === null) {
      return (
<<<<<<< HEAD:src/components/Order/OrderMain/OrderState/OrderReleased/OrderReleased.js
        <div className={styles.container}>
          <h2 className={styles.title}>Processing withdrawal...</h2>
=======
       <I18n ns="translations">
	   {(t) => (
        <div className="col-xs-12 text-center order-status-section">
          <h2 style={{ margin: '0' }}>{t('order.processing')}</h2>
>>>>>>> Translated further components:src/containers/order/OrderReleased.js
          <OrderLinks {...this.props} />
        </div>
		)}</I18n>
      );
    }

    return (
<<<<<<< HEAD:src/components/Order/OrderMain/OrderState/OrderReleased/OrderReleased.js
      <div className={styles.container}>
        <h2 className={styles.title}>
          Funds released, awaiting confirmations ({this.tx.confirmations}/{this.minConfirmations})
        </h2>
        <h3 className={styles.subtitle}>
          Transaction ID:{' '}
          <a href={getBlockchainUrl(this.coin.code, this.txId)} target="_blank" className="text-green">
=======
	<I18n ns="translations">
	{(t) => (
      <div className="col-xs-12 text-center order-status-section">
        <h2 style={{ margin: '0' }}>
          {t('order.released')} ({this.tx.confirmations}/{
            this.minConfirmations
          })
        </h2>
        <h5>
          {t('order.txid')}:{' '}
          <a
            href={getBlockchainUrl(this.coin.code, this.txId)}
            target="_blank"
            className="text-green"
          >
>>>>>>> Translated further components:src/containers/order/OrderReleased.js
            {this.txId}
          </a>
        </h3>

        <OrderLinks coin={this.coin.code} txId={this.txId} {...this.props} />
      </div>
	)}</I18n>
    );
  }
}

export default OrderReleased;
