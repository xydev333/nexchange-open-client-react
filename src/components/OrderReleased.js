import React, { Component } from 'react';
import moment from 'moment';
import config from '../config';


class OrderReleased extends Component {
	constructor(props) {
		super(props);

		this.state = {
			countdown: '...'
		}

		this.estimateCountdown = this.estimateCountdown.bind(this);
		this.setVariables = this.setVariables.bind(this);

		this.setVariables(props);
	}

	componentDidMount() {
		this.interval = setInterval(() => {
			this.setState({
				countdown: this.state.countdown - 1000
			})
		}, 1000);

		this.estimateCountdown(this.props);
	}

	setVariables(props) {
		this.coin = props.order.pair.base;
		this.coinName = this.coin.code;
		this.minConfirmations = this.coin.min_confirmations;
		this.tx = props.order.transactions[1];
		this.txId = this.tx.tx_id;

		if (this.coinName == 'BTC') this.confirmationWaitTime = 600000;
		else if (this.coinName == 'LTC') this.confirmationWaitTime = 150000; // 2.5mins
		else if (this.coinName == 'ETH') this.confirmationWaitTime = 60000; // ETH, 0.2mins

		this.allConfirmationsWaitTime = this.confirmationWaitTime * this.minConfirmations;

		if (this.coinName == 'ETH') this.blockchainUrl = `https://etherscan.io/tx/${this.txId}`;
		else if (this.coinName == 'LTC') this.blockchainUrl = `https://live.blockcypher.com/ltc/tx/${this.txId}/`;
		else if (this.coinName == 'BTC') this.blockchainUrl = `https://live.blockcypher.com/btc/tx/${this.txId}/`;
	}

	estimateCountdown(props) {
		let estimate = this.allConfirmationsWaitTime - (this.allConfirmationsWaitTime * (this.tx.confirmations/this.minConfirmations));

		this.setState({
			countdown: estimate
		});
	}

	componentWillReceiveProps(nextProps) {
		this.setVariables(nextProps);
		this.estimateCountdown(nextProps);
	}

	componentWillUnmount() {
		clearInterval(this.interval);
	}

	render() {
		return (
			<div className="col-xs-12 text-center order-status-section">
				<h2 style={{margin: "0"}}>Funds released, awaiting confirmations ({this.tx.confirmations}/{this.minConfirmations})</h2>
				<h5>Transaction ID: <a href={this.blockchainUrl} target="_blank" className="text-green">{this.txId}</a></h5>

				{this.state.countdown >= 0 ? (
					<h5>Estimated time left until all confirmations: <b className="text-green">{moment.utc(this.state.countdown).format('mm:ss')}</b></h5>
				) : (
					<h5>The transaction should have gotten the required number of confirmation by now.</h5>					
				)}

				<a href={`${config.API_BASE_URL}/orders/${this.props.orderRef}?format=json`} target="_blank"><h4 style={{margin: "25px 0 0px", "fontWeight": "500"}}>See your order details on our API</h4></a>
				<a href={this.blockchainUrl} target="_blank"><h4 style={{margin: "5px 0 18px", "fontWeight": "500"}}>See your order details on blockchain</h4></a>
			</div>
		)
	}

};

export default OrderReleased;
