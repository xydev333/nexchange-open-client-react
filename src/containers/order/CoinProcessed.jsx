import React, { Component } from 'react';
import Helpers from '../../helpers';

class CoinProcessed extends Component {
  constructor(props) {
    super(props);
    this.state = { order: this.props.order };
    this.prepareState = this.prepareState.bind(this);
  }

  componentDidMount() {
    this.prepareState(this.props);
  }

  componentWillReceiveProps(nextProps) {
    this.setState({ order: nextProps.order }, () => {
      this.prepareState(nextProps);
    });
  }

  prepareState(props) {
    if (props.type === 'Deposit') {
      this.setState({
        coin: props.order.pair.quote.code,
        oppositeCoin: props.order.pair.base.code,
        amount: parseFloat(props.order.amount_quote),
        address: props.order.deposit_address
          ? props.order.deposit_address.address
          : '',
        order: props.order,
      });
    } else if (props.type === 'Receive') {
      this.setState({
        coin: props.order.pair.base.code,
        oppositeCoin: props.order.pair.quote.code,
        amount: parseFloat(props.order.amount_base),
        address: props.order.withdraw_address
          ? props.order.withdraw_address.address
          : '',
        order: props.order,
      });
    }
  }

  render() {
    let rates = ``;

    if (this.state.order && this.state.order.price) {
      rates += `Rates at order creation: \n`;
      rates += `1 ${this.state.coin} = `;

      if (this.props.type === 'Deposit')
        rates += `${(1 / this.state.order.price.rate).toFixed(8)} ${
          this.state.oppositeCoin
        }\n`;
      else if (this.props.type === 'Receive')
        rates += `${this.state.order.price.rate.toFixed(8)} ${
          this.state.oppositeCoin
        }\n`;

      rates += `1 ${this.state.coin} = `;

      if (this.props.type === 'Deposit')
        rates += `${(
          1 /
          this.state.order.price.rate *
          this.state.order.price.rate_usd
        ).toFixed(8)} USD\n`;
      else if (this.props.type === 'Receive')
        rates += `${this.state.order.price.rate_usd.toFixed(8)} USD\n`;

      rates += `1 ${this.state.coin} = `;

      if (this.props.type === 'Deposit')
        rates += `${(
          1 /
          this.state.order.price.rate *
          this.state.order.price.rate_btc
        ).toFixed(8)} BTC`;
      else if (this.props.type === 'Receive')
        rates += `${this.state.order.price.rate_btc.toFixed(8)} BTC`;

      if (
        this.state.order.user_provided_amount === 1 &&
        this.props.type === 'Receive'
      ) {
        rates += `\n\nWithdrawal fee: \n`;
        rates += `${this.state.order.withdrawal_fee} ${
          this.state.order.pair.base.code
        }`;
      } else if (
        this.state.order.user_provided_amount === 0 &&
        this.props.type === 'Deposit'
      ) {
        rates += `\n\nWithdrawal fee: \n`;
        rates += `${this.state.order.withdrawal_fee_quote} ${
          this.state.order.pair.quote.code
        }`;
      }
    }

    return (
      <div className="col-xs-12 col-sm-6">
        <div
          className={`coin-box box media ${
            this.props.type === 'Deposit' &&
            Helpers.isFiatOrder(this.props.order)
              ? 'fiat'
              : ''
          }`}
        >
          <div className="media-left">
            <i
              className={`coin-icon cc-${this.state.coin} ${this.state.coin}`}
            />
          </div>

          <div className="media-body">
            <h5>
              <b>
                {this.props.type} {this.state.amount} {this.state.coin}
              </b>

              <i
                className="fa fa-question-circle"
                data-toggle="tooltip"
                data-placement="top"
                style={{ marginLeft: 8 }}
                data-original-title={rates}
              />
            </h5>
            <h6>{this.state.address}</h6>
          </div>
        </div>
      </div>
    );
  }
}

export default CoinProcessed;
