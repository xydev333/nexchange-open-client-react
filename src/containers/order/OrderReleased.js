import React, { Component } from 'react';
import { I18n } from 'react-i18next';
import _ from 'lodash';
import getBlockchainUrl from '../../helpers/getBlockchainUrl';
import OrderLinks from './OrderLinks';

class OrderReleased extends Component {
  constructor(props) {
    super(props);

    this.coin = props.order.pair.base;
    this.minConfirmations = this.coin.min_confirmations;
    this.tx = _.find(props.order.transactions, { type: 'W' });
    this.txId = this.tx.tx_id;
  }

  UNSAFE_componentWillReceiveProps(nextProps) {
    this.tx = _.find(nextProps.order.transactions, { type: 'W' });
    this.txId = this.tx.tx_id;
  }

  render() {
    if (this.txId === '' || this.txId === null) {
      return (
       <I18n ns="translations">
	   {(t) => (
        <div className="col-xs-12 text-center order-status-section">
          <h2 style={{ margin: '0' }}>{t('order.processing')}</h2>
          <OrderLinks {...this.props} />
        </div>
		)}</I18n>
      );
    }

    return (
	<I18n ns="translations">
	{(t) => (
      <div className="col-xs-12 text-center order-status-section">
        <h2 style={{ margin: '0' }}>
          {t('order.released')} ({this.tx.confirmations}/{
            this.minConfirmations
          })
        </h2>
        <h5>
          {t('order.txid')}:{' '}
          <a
            href={getBlockchainUrl(this.coin.code, this.txId)}
            target="_blank"
            className="text-green"
          >
            {this.txId}
          </a>
        </h5>

        <OrderLinks coin={this.coin.code} txId={this.txId} {...this.props} />
      </div>
	)}</I18n>
    );
  }
}

export default OrderReleased;
