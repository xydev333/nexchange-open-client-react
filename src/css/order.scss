@import "variables";
@import "mixins";
@import "bootstrap/variables";
@import "bootstrap/mixins";

#order { background: #f8f7f8; padding: 15px 0 0 0;
	a { color: $green; 
		&:visited, &:focus { text-decoration: none; }
		&:hover { text-decoration: underline; }
		&.btn { color: #fff;
			&:hover { text-decoration: none; }
		}
	}
	#order-ref { margin-bottom: 15px; }
	.box {
		h4 { font-size: 15px; }
		h5, h6 { margin: 0; text-transform: none; }
		.coin-icon { position: relative; top: 8px; font-size: 28px;
			@media (min-width: $screen-sm-min) { top: 9px; }
			@media (min-width: $screen-md-min) { top: 11px; }
		}
		.text-warning { color: $green; }		

		@media (min-width: $screen-sm-min) { margin-bottom: 30px; }
	}

	&.order-fiat {
		.box.fiat {
			h5 { margin-top: 11px; padding-bottom: 7px;
				@media (min-width: $screen-sm-min) { margin-top: 11px; padding-bottom: 13px; }
			}
		}
	}

	.coin-box { padding-left: 10px;
		.tooltip-inner { text-align: left; }
	}
	#time-remaining { color: #2cb4a0; }
	.order-status-section { 
		h2 { margin-top: 0; }
		.explanation { font-size: 16px; margin-top: 15px; }
	}


	@media (min-width: $screen-sm-min) { padding: 25px 0;
		h5 { font-size: 15px; }
		h6 { font-size: 12px; }
		.box h4 { font-size: 18px; }
	}

	@media (min-width: $screen-md-min) {
		h5 { font-size: 17px; }
		h6 { font-size: 16px; }
	}
}

#order-header {
	#bookmark-button { font-size: 20px; padding: 0; margin-top: 0; color: $green; }

	@media (min-width: $screen-ms-min) {
		h3 { float: left; }
		#bookmark-button { float: right; margin-top: 22px; }
	}
}

#order-payment { text-align: center;
	h3 { margin-top: 0; }
	@media (min-width: $screen-ms-min) { text-align: left; }
}

#bookmark {
	.bookmark-text{ font-size: 19px; }
	.input-group-addon { cursor: pointer; padding-top: 0;
		i { font-size: 18px; }
	}
	.form-group { margin-top: 10px; }
	input { font-size: 18px; }
}

.order-status-section {
	h2 { font-size: 24px; }

	@media (min-width: $screen-sm-min) {
		h2 { font-size: 30px; }
	}
}

#share-referral, #notifications { text-align: center;
	.box { padding: 20px; }
	.share { margin-top: 15px; }
	.fa { font-size: 45px; margin: 0 15px 0 0; color: #d6d6d6; }
	h2 { margin-top: 0; font-size: 22px;
		@media (min-width: $screen-sm-min) { font-size: 26px; }
		@media (min-width: $screen-md-min) { font-size: 34px; }
	}
}

#notifications {
	.form-group { margin-top: 10px; }
	button { width: 100%; margin-bottom: 0; }
	form { padding-top: 15px;
		h4 { position: absolute; top: -15px; left: 50%; transform: translate(-50%, 0); }
	}
	@media (min-width: $screen-sm-min) { button { width: auto; } }
}

@include keyframes(color-change) {
	from { color: #adadad; }
	to { color: #fbc02d; } 
}

@import "order-status";
@import "referral";
