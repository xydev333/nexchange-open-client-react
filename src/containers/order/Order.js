import React, { Component } from 'react';
import { connect } from 'react-redux';

import { fetchOrder } from '../../actions';

import isFiatOrder from '../../helpers/isFiatOrder';
import config from '../../config';

import '../../css/order.scss';

import OrderInfo from './OrderInfo';
import OrderTop from './OrderTop';
import CoinProcessed from './CoinProcessed';

import NotFound from '../../components/NotFound';
import Loading from '../../components/Loading';

import Notifications from '../../containers/Notifications';
import RefundAddress from '../../containers/RefundAddress';
import ReferralBox from '../../containers/ReferralBox';

class Order extends Component {
  constructor(props) {
    super(props);

    if (this.props.order && this.props.match.params.orderRef === this.props.order.unique_reference) {
      this.state = { order: this.props.order };
    } else {
      this.state = {};
    }
  }

  componentDidMount() {
    this.props.fetchOrder(this.props.match.params.orderRef);
  }

  componentDidUpdate(prevProps) {
    if (this.props.location !== prevProps.location) {
      clearTimeout(this.timeout);
      clearInterval(this.interval);
      this.props.fetchOrder(this.props.match.params.orderRef);
    }
  }

  componentWillUnmount() {
    clearInterval(this.interval);
    clearTimeout(this.timeout);
  }

  UNSAFE_componentWillReceiveProps(nextProps) {
    this.setState({ order: nextProps.order });

    this.timeout = setTimeout(() => {
      this.props.fetchOrder(this.props.match.params.orderRef);
    }, config.ORDER_DETAILS_FETCH_INTERVAL);

    if (nextProps.order !== 429) {
      this.setState({ order: nextProps.order });

      if (
        this.props.order &&
        this.props.order.status_name.length > 0 &&
        this.props.order.status_name[0][0] === 11 &&
        nextProps.order.status_name[0][0] === 12
      ) {
        window.ga('send', 'event', 'Order', 'order paid', nextProps.unique_reference);
      }
    }
  }

  render() {
    if (this.state.order == null) {
      return <Loading />;
    } else if (this.state.order === 404) {
      return <NotFound />;
    } else if (typeof this.state.order === 'object') {
      return (
        <div id="order" className={isFiatOrder(this.state.order) ? 'order-fiat' : 'order-crypto'}>
          <div className="container">
            <OrderTop order={this.state.order} />

            <div className="row">
              <CoinProcessed type="Deposit" order={this.state.order} />
              <CoinProcessed type="Receive" order={this.state.order} />

              <OrderInfo order={this.state.order} />
              <Notifications order={this.state.order} />

              {!isFiatOrder(this.state.order) && <RefundAddress order={this.state.order} />}

              <ReferralBox order={this.state.order} />
            </div>
          </div>
        </div>
      );
    }
  }
}

const mapStateToProps = ({ order }) => ({ order });

export default connect(mapStateToProps, { fetchOrder })(Order);
