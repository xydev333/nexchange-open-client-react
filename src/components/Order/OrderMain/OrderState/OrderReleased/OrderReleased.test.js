import React from 'react';
import { shallow } from 'enzyme';
import OrderReleased from './OrderReleased';
import orderNoTx from 'Mocks/orderNoTx';
import order from 'Mocks/order';

describe('OrderReleased', () => {
  it('renders correctly', () => {
    const wrapShallow = shallow(<OrderReleased order={orderNoTx} />);
    expect(wrapShallow).toMatchSnapshot();
    wrapShallow.setProps({ order });
    expect(wrapShallow).toMatchSnapshot();
  });
});
