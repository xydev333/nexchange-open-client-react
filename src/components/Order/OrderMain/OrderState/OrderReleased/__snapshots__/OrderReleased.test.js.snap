// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderReleased renders correctly 1`] = `ShallowWrapper {}`;

exports[`OrderReleased renders correctly 2`] = `ShallowWrapper {}`;
